src/Algebra/Action/Construct/FreeMonoid.agda

```diff
@@ -0,0 +1,82 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The free MonoidAction on a SetoidAction, arising from ListAction
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Relation.Binary.Bundles using (Setoid)
+
+module Algebra.Action.Construct.FreeMonoid
+  {a c r ℓ} (M : Setoid c ℓ) (S : Setoid a r)
+  where
+
+open import Algebra.Action.Bundles
+open import Algebra.Action.Structures using (IsLeftAction; IsRightAction)
+open import Algebra.Bundles using (Monoid)
+open import Algebra.Structures using (IsMonoid)
+
+open import Data.List.Base using (List; []; _++_)
+import Data.List.Properties as List
+import Data.List.Relation.Binary.Equality.Setoid as ≋
+open import Data.Product.Base using (_,_)
+
+open import Function.Base using (_∘_)
+
+open import Level using (Level; _⊔_)
+
+
+------------------------------------------------------------------------
+-- First: define the Monoid structure on List M.Carrier
+-- NB should be defined somewhere under `Data.List`!?
```

Comment:
Data.List.Action.Construct.FreeMonoid ?

Reply:
Interesting suggestion!
But NB. as with #2360 suggest that this be dealt with as a separate issue/PR. 
The code here is self-contained, so can be lifted out as a downstream refactoring?

Reply:
Agreed, that can be a separate PR.

Reply:
Suggest: `Data.List.Relation.Binary.Equality.Setoid.Properties`? which does not currently exist... but conforms to other (pre-)existing nodes in the `Data.*.Relation.Binary.*.Setoid.*` style?